gstreamer-plugin/git-patch/0001-svthevcenc-Add-svthevc-encoder-element.patch

```diff
@@ -0,0 +1,2123 @@
++{
++  GstVideoEncoderClass parent_class;
++};
++
++GType gst_svthevc_enc_get_type (void);
++
++G_END_DECLS
++#endif /* __GST_SVTHEVC_ENC_H__ */
+diff --git a/ext/svthevc/meson.build b/ext/svthevc/meson.build
+new file mode 100644
+index 0000000..5fdef45
+--- /dev/null
++++ b/ext/svthevc/meson.build
+@@ -0,0 +1,16 @@
++svthevc_sources = [
++  'gstsvthevcenc.c',
++]
++
++svthevc_dep = dependency('SvtHevcEnc', required: false)
++
++if svthevc_dep.found()
++  gstsvthevc = library('gstsvthevc',
++    svthevc_sources,
++    c_args : gst_plugins_bad_args,
++    include_directories : [configinc],
++    dependencies : [gstbase_dep, gstpbutils_dep, gstvideo_dep, svthevc_dep],
++    install : true,
++    install_dir : plugins_install_dir,
++  )
++endif
+diff --git a/tests/check/Makefile.am b/tests/check/Makefile.am
+index 163ef06..8c5456d 100644
+--- a/tests/check/Makefile.am
++++ b/tests/check/Makefile.am
+@@ -216,6 +216,12 @@ else
+ check_msdk=
+ endif
+ 
++if USE_SVTHEVC
++check_svthevcenc=elements/svthevcenc
++else
++check_svthevcenc=
++endif
++
+ VALGRIND_TO_FIX = \
+ 	elements/mpeg2enc \
+ 	elements/mplex    \
+@@ -297,6 +303,7 @@ check_PROGRAMS = \
+ 	$(check_player) \
+ 	$(check_webrtc) \
+ 	$(check_msdk) \
++	$(check_svthevcenc) \
+ 	$(EXPERIMENTAL_CHECKS)
+ 
+ noinst_HEADERS = elements/mxfdemux.h libs/isoff.h
+diff --git a/tests/check/elements/.gitignore b/tests/check/elements/.gitignore
+index 5265a88..add5ee8 100644
+--- a/tests/check/elements/.gitignore
++++ b/tests/check/elements/.gitignore
+@@ -65,4 +65,5 @@ voaacenc
+ voamrwbenc
+ webrtcbin
+ x265enc
++svthevcenc
+ zbar
+diff --git a/tests/check/elements/svthevcenc.c b/tests/check/elements/svthevcenc.c
+new file mode 100644
+index 0000000..f0c1631
+--- /dev/null
++++ b/tests/check/elements/svthevcenc.c
+@@ -0,0 +1,228 @@
++/* GStreamer
++ * Copyright (C) 2019 Yeongjin Jeong <yeongjin.jeong@navercorp.com>
++ *
++ * This library is free software; you can redistribute it and/or
++ * modify it under the terms of the GNU Library General Public
++ * License as published by the Free Software Foundation; either
++ * version 2 of the License, or (at your option) any later version.
++ *
++ * This library is distributed in the hope that it will be useful,
++ * but WITHOUT ANY WARRANTY; without even the implied warranty of
++ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
++ * Library General Public License for more details.
++ *
++ * You should have received a copy of the GNU Library General Public
++ * License along with this library; if not, write to the
++ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
++ * Boston, MA 02110-1301, USA.
++ */
++
++#include <gst/check/gstcheck.h>
++
++static GstStaticPadTemplate sinktemplate = GST_STATIC_PAD_TEMPLATE ("sink",
++    GST_PAD_SINK,
++    GST_PAD_ALWAYS,
++    GST_STATIC_CAPS ("video/x-h265, "
++        "width = (int) [1, MAX], "
++        "height = (int) [1, MAX], " "framerate = (fraction) [0, MAX]"));
```

Comment:
width max: 8192, height max: 4320
min should at least be a macro block size, that is 64.

Reply:
I fixed this.